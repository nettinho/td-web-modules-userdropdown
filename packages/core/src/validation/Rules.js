import _ from "lodash/fp";
import * as ErrorMessages from "./ErrorMessages.js";

export const required = text => {
  if (_.isEmpty(text)) {
    return ErrorMessages.isRequired;
  } else {
    return null;
  }
};

export const mustMatch = (name, nameText) => {
  return (text, state) => {
    return state[name] === text ? null : ErrorMessages.mustMatch(nameText);
  };
};

export const minLength = length => {
  return text => {
    return text.length >= length ? null : ErrorMessages.minLength(length);
  };
};

export const validEmail = email => {
  const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  if (re.test(email)) {
    return null;
  } else {
    return ErrorMessages.invalidFormat("my@email.com");
  }
};

export const validUrls = urls => {
  if (!_.isUndefined(urls)) {
    for (let url of urls) {
      if (_.isEmpty(url.url_name) || _.isEmpty(url.url_value)) {
        return ErrorMessages.isRequired;
      }
    }
  }
  return null;
};
