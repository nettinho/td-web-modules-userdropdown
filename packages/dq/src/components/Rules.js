import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Header, Icon, Table, Segment } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import RuleRow from "./RuleRow";
import RulesActions from "./RulesActions";

export const QualityHeader = () => (
  <Header as="h2">
    <Icon circular name="check" />
    <Header.Content>
      <FormattedMessage id="quality.header" />
      <Header.Subheader>
        <FormattedMessage id="quality.subheader" />
      </Header.Subheader>
    </Header.Content>
  </Header>
);

export const Rules = ({ rules }) => (
  <React.Fragment>
    <Segment>
      <QualityHeader />
      <RulesActions />
      <Table celled striped compact selectable>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell
              content={<FormattedMessage id="quality.business_concept" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.name" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.description" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.goal" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.minimum" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.principle" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.type" />}
            />
            <Table.HeaderCell
              content={<FormattedMessage id="quality.type_params" />}
            />
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {rules.map((s, i) => <RuleRow key={i} {...s} />)}
        </Table.Body>
      </Table>
    </Segment>
  </React.Fragment>
);

Rules.propTypes = {
  rules: PropTypes.array
};

const mapStateToProps = ({ rules }) => ({
  rules
});

export default connect(mapStateToProps)(Rules);
