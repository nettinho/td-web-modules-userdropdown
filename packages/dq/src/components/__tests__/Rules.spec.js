import React from "react";
import { shallow } from "enzyme";
import { Rules } from "../Rules";

describe("<Rules />", () => {
  it("matches the latest snapshot", () => {
    const rules = {
      rules: [
        {
          id: 123,
          business_concept_id: "2D2B3",
          name: "control1",
          description: "desc1",
          goal: 10,
          minimum: 1,
          principle: { name: "principle1" },
          type: "type1",
          type_params: {},
          status: "status1"
        },
        {
          id: 122,
          business_concept_id: "2D2B4",
          name: "control2",
          description: "desc2",
          goal: 10,
          minimum: 1,
          principle: { name: "principle2" },
          type: "type2",
          type_params: {},
          status: "status2"
        }
      ]
    };
    const wrapper = shallow(<Rules {...rules} />);
    expect(wrapper).toMatchSnapshot();
  });
});
