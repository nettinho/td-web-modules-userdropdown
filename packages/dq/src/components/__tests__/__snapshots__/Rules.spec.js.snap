// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rules /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Segment>
    <QualityHeader />
    <RulesActions />
    <Table
      as="table"
      celled={true}
      compact={true}
      selectable={true}
      striped={true}
    >
      <TableHeader
        as="thead"
      >
        <TableRow
          as="tr"
          cellAs="td"
        >
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.business_concept"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.name"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.description"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.goal"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.minimum"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.principle"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.type"
                values={Object {}}
              />
            }
          />
          <TableHeaderCell
            as="th"
            content={
              <FormattedMessage
                id="quality.type_params"
                values={Object {}}
              />
            }
          />
        </TableRow>
      </TableHeader>
      <TableBody
        as="tbody"
      >
        <withRouter(RuleRow)
          business_concept_id="2D2B3"
          description="desc1"
          goal={10}
          id={123}
          key="0"
          minimum={1}
          name="control1"
          principle={
            Object {
              "name": "principle1",
            }
          }
          status="status1"
          type="type1"
          type_params={Object {}}
        />
        <withRouter(RuleRow)
          business_concept_id="2D2B4"
          description="desc2"
          goal={10}
          id={122}
          key="1"
          minimum={1}
          name="control2"
          principle={
            Object {
              "name": "principle2",
            }
          }
          status="status2"
          type="type2"
          type_params={Object {}}
        />
      </TableBody>
    </Table>
  </Segment>
</React.Fragment>
`;
